### PR TITLE
Fix session control window error handling

### LINKED ISSUE #2768: Viewer context switching breaks on CORS issues
Catching the expected security issue when an open session (with a different origin) should not result in creating a new session.

https://github.com/comic/grand-challenge.org/blob/fd32bc360bc60aa99348efce3a343c39ab6e9017/app/grandchallenge/workstations/static/workstations/js/session-control.mjs#L30

### PR DESCRIPTION
**Pull request by chrisvanrun**
Closes #2768 

Fix non-homologous-origin session window always resulting in fallback session creation.

### CODE REVIEW

**Comment by chrisvanrun** [comment]
Anne mentioned that we should likely keep throwing any unexpected errors to Sentry, so I redid the logic a bit.

**Comment by chrisvanrun** [comment]
Tests failed on:
- #2759

**Review by HarmvZ — CHANGES_REQUESTED** [review]
Finally block returning early as discussed in call.

File: app/grandchallenge/workstations/static/workstations/js/session-control.mjs [hunk 1]
```diff
@@ -121,6 +123,9 @@ function setSpinner(element) {
 
 function removeSpinner(element) {
     const spinner = element.querySelector(".spinner-border");
+    if (spinner === null) {
+
+    }
```
Comment (HarmvZ) [inline app/grandchallenge/workstations/static/workstations/js/session-control.mjs:128]: Leftover?

```suggestion
```

**Review by HarmvZ — CHANGES_REQUESTED** [review]

File: app/grandchallenge/workstations/static/workstations/js/session-control.mjs (hunk 1 above)
Comment (chrisvanrun) [reply app/grandchallenge/workstations/static/workstations/js/session-control.mjs]: Not totally, I got disrupted in person when working on this and completely forgot to finish the copy-paste =/. I noticed that running removeSpinner multiple times doesn't work, this is a fix.

**Review by HarmvZ — APPROVED** [review]

**Pull request merged**